src/main/java/reactor/netty/resources/PooledConnectionProvider.java

```diff
@@ -531,6 +531,10 @@ public void onUncaughtException(Connection connection, Throwable error) {
 		@Override
 		public void onStateChange(Connection connection, State newState) {
 			if (newState == State.CONFIGURED) {
+				// First send a notification that the connection is ready and then change the state
+				// In case a cancellation was received, ChannelOperations will be disposed
+				// and there will be no subscription to the I/O handler at all.
+				// https://github.com/reactor/reactor-netty/issues/1165
```

Comment:
it seems the existing code was already doing the right thing (sink.success then obs.onStateChange), but since nothing changed here other than the comment, I have to ask: is this comment on purpose?

Reply:
yes it is there to remind that `success` is first and then `observe.onStateChange`